# Exercise 03: Probability Analysis — NCII Content Moderation
# Cluster 3 - Statistics and Probability
# Framework: EU AI Act Article 52, NIST AI RMF Measure
# Case Study: NCII Platform Governance

In [1]:
import numpy as np

# ContentGuard removes NCII correctly 70% of the time
# That means it FAILS 30% of the time

p_fail = 0.30
p_success = 0.70

print(f"Probability of failing to remove NCII: {p_fail*100}%")
print(f"Probability of correctly removing NCII: {p_success*100}%")

Probability of failing to remove NCII: 30.0%
Probability of correctly removing NCII: 70.0%


In [8]:
p_fail_twice= p_fail**2
p_fail_thrice=p_fail**3


print("Probability of failing once is = ",p_fail)
print("Probability of failing twice in a row = ",p_fail_twice)
print("Probability of failing three times in a row = ",p_fail_thrice)
print(f"If 1000 women report daily, how many experience all reports ignored for 3 reports = { p_fail_thrice*1000:.0f}")

Probability of failing once is =  0.3
Probability of failing twice in a row =  0.09
Probability of failing three times in a row =  0.026999999999999996
If 1000 women report daily, how many experience all reports ignored for 3 reports = 27


In [11]:
# Required failure rate to protect fewer than 5 women per day
required_p = (5/1000) ** (1/3)

print(f"Required failure rate: {required_p*100:.1f}%")
print(f"ContentGuard current failure rate: {p_fail*100:.0f}%")

if p_fail > required_p:
    print("ContentGuard FAILS to meet this standard")
else:
    print("ContentGuard meets this standard")

Required failure rate: 17.1%
ContentGuard current failure rate: 30%
ContentGuard FAILS to meet this standard


**Exercise 03: Probability Analysis — NCII Content Moderation — Governance Reflection**

Mean and standard deviation describe how data is distributed within a dataset — one shows central tendency, the other shows spread. But they don't answer the governance question: "How many real people are harmed?" Probability analysis bridges this gap. A 2.7% failure rate sounds acceptable until you multiply it by scale: 1000 women report NCII daily, 2.7% means 27 women every single day have all three reports ignored by ContentGuard. Small percentages become large harms at population scale.

ContentGuard currently fails to remove NCII 30% of the time. To reduce daily harm to fewer than 5 women, it must achieve a failure rate below 17.1%. Currently it falls 12.9 percentage points short of this standard. This gap is not acceptable under EU AI Act governance requirements.

EU AI Act Article 52 requires platforms to label AI-generated content as artificial and implement safeguards against non-consensual intimate imagery. However, the regulation has a critical gap — it does not explicitly define what failure rates are acceptable for NCII detection systems. This is where probability analysis becomes essential governance practice: by quantifying failure rates and translating them into real-world harm counts, auditors and regulators can demand specific performance standards. ContentGuard must either improve to <17.1% failure rate or disclose to users that an average of 27 NCII reports daily will be missed.

Under India's DPDP Act 2023, platforms processing personal data in content moderation decisions must demonstrate fairness and explain why automated decisions fail. A 30% failure rate on NCII — disproportionately harming women — would likely constitute a breach of DPDP's fairness principle and require human review override mechanisms.
